packages/lib/src/vis/shared/Annotation.tsx

```diff
@@ -15,12 +14,13 @@ interface Props extends HTMLAttributes<HTMLDivElement> {
 function Annotation(props: Props) {
   const { x, y, scaleOnZoom, children, style, ...divProps } = props;
 
-  const camera = useThree((state) => state.camera);
-
-  const { dataToWorld, worldToHtml } = useAxisSystemContext();
-  const htmlPt = worldToHtml(dataToWorld(new Vector2(x, y)));
-
-  useFrameRendering();
+  const { htmlPt, cameraScale } = useCameraState(
+    (camera, context, x, y) => ({
+      htmlPt: grabHtmlPoint(camera, context, new Vector2(x, y)),
+      cameraScale: grabCameraScale(camera),
+    }),
+    [x, y]
+  );
```

Comment:
In this case, the grabber is declared inline in order to compose the result of two grabbers into a single camera state. The reason for this is that we want to avoid multiple calls to `useCameraState` in the same component, as this would most likely lead to two consecutive renders on (potentially) every frame.

Because the grabber is recreated on every frame, the `useEffect` inside `useCameraState` ends up being called on every render, which basically means that the grabber is called twice in a row on every frame. The second time it's called, though, the result is the same, so the deep comparison check ensures that we don't re-render again (and therefore get stuck in a render loop).

To work around this unnecessary deep comparison, we could declare the grabber outside of the component and import it (like we do with other grabbers like `grabHtmlSelection`). This would work in the example above, but the resulting grabber would be used only by `Annotation`, so it feels a little unnecessary... Especially since the grabber is not computationally expensive.

If the extra deep comparison check is a problem and moving the grabber outside of the component is not possible (e.g. because it makes use of a variable that is declared inside the component), then it's always possible to wrap it in a `useCallback`:

```ts
const { x, y } = props;

const grabber = useCallback((camera, context) => ({
  htmlPt: grabHtmlPoint(camera, context, new Vector2(x, y)),
  cameraScale: grabCameraScale(camera),
}), [x, y]); // `x` and `y` are referenced directly instead of via the grabber arguments array of `useCameraState`

const { htmlPt, cameraScale } = useCameraState(grabber);
```

Now, you may be wondering why I chose to put the `grabber` function as dependency of the `useEffect` in `useCameraState`. Well, it's simply to avoid ending up with obscure bugs when misusing `useCameraState`. Typically, the code below works perfectly fine even though the grabber is declared inline and references variables declared in the component instead of passing them as grabber arguments:

```ts
const { x, y } = props;

const { htmlPt, cameraScale } = useCameraState(((camera, context) => ({
  htmlPt: grabHtmlPoint(camera, context, new Vector2(x, y)),
  cameraScale: grabCameraScale(camera),
}); // no grabber arguments array
```

If I remove the `grabber` function from the `useEffect`'s dependencies, then any changes to the `x` and `y` props will not be detected and the camera state will not be updated. We could try to prevent code like this from being written by configuring the `exhaustive-deps` linting rule to recognise `useCameraState` (https://www.npmjs.com/package/eslint-plugin-react-hooks), but this would only work in this project -- consumers of @h5web/lib would have to configure the linting rule in their own projects as well...

In summary:

- `useCameraState` is very robust and can be used in many ways; 
- some of these ways are not quite optimal, but their performance impacts are typically minimal;
- if performance is a worry (i.e. if the grabber is computationally expensive), an optimal solution is available.